wallet/wallet.c

```diff
@@ -667,6 +667,27 @@ bool wallet_channels_load_active(const tal_t *ctx, struct wallet *w)
 	return ok;
 }
 
+/* Upgrade of db (or initial create): do we have anything to scan for? */
+static bool wallet_ever_used(struct wallet *w)
+{
+	sqlite3_stmt *stmt;
+	bool channel_utxos;
+
+	/* If we ever handed out an address. */
+	if (db_get_intvar(w->db, "bip32_max_index", 0) != 0)
+		return true;
+
+	/* Or we could have had a channel terminate unilaterally,
```

Comment:
How about, if we are a fresh, satoshiless node, somebody opens a channel to us, gives us money, then closes the channel bilaterally?  Also, which unilateral is this?

Reply:
This would be the their_unilateral/to_us output, since it's the only case in which we don't have the secret key integrated into the HD key structure, instead it's a collaboratively generated key. In all other cases `onchaind` will take care to respend to a key in the hierarchy.

Reply:
okay, thank you.